src/main/webapp/app/overview/exercise-chatbot/chatbot-popup/chatbot-popup.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, OnInit } from '@angular/core';
+import { AccountService } from 'app/core/auth/account.service';
+import { User } from 'app/core/user/user.model';
+import { Subscription, tap } from 'rxjs';
+
+@Component({
+    selector: 'jhi-chatbot-popup',
+    templateUrl: './chatbot-popup.component.html',
+    styleUrls: ['./chatbot-popup.component.scss'],
+})
+export class ChatbotPopupComponent implements OnInit {
+    public firstName: string | undefined;
+
+    private authStateSubscription: Subscription;
+
+    constructor(private accountService: AccountService) {}
+
+    ngOnInit() {
+        this.authStateSubscription = this.accountService
+            .getAuthenticationState()
```

Comment:
Not sure, but probably get authentication state is used to keep track of login/logout events. What you actually need is

```suggestion
            this.accountService.identity().then((user: User) => {
            this.user = user!;
        });
```

Reply:
I updated it, now works with your suggestion